test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -450,6 +452,19 @@ def setUpClass(cls):
         cls.check_users_exist()
         cls.init_servers()
         if cls.use_cur_setup:
+            test_user = pwd.getpwuid(os.getuid())[0] + '@*'
+            cls.users = {test_user: "managers",
+                         str(MGR_USER) + '@*': "managers",
+                         str(OPER_USER) + '@*': "operators"}
+            server_stat = cls.server.status(SERVER, ["managers", "operators"])
+            if len(server_stat) > 0:
+                server_stat = server_stat[0]
+            for user, role in cls.users.items():
+                if role not in server_stat or user not in server_stat[role]:
+                    attr = {role: (INCR, user)}
+                    cls.server.manager(MGR_CMD_SET, SERVER, attr, sudo=True)
```

Comment:
Also, these added mgrs and oprs will be saved in setup (which is not part of the original setup which user asked to save and use) so shouldn't we need to take care of removing added mgrs and oprs while restoring saved setup?

Reply:
I agree with you. If I remove the added mgrs and oprs in teardown(while restoring the saved setup), I may have to add it again in the next testcase's setup. so even though I have added it in setupclass I will have to add it in setup as well. Does that sound fine?

Reply:
On second thought (as per our offline skype chat): I think we should move this setting mgrs and oprs into `save_configuration` and reverting it to `load_configuration`, that way if someone else other than `setUp/tearDown(class)` is using save/load configuration then they will be also beneficial for this change.

Reply:
Thank you @hirenvadalia for the suggestion. I am making the change.

Reply:
@hirenvadalia As per your suggestion, I am using only one variable for the fix. Please look at it.

Also, I tried  adding the managers and operators in `save_configuration` and `load configuration`. It works upto the point where `load_configuration` is called in teardown. I am removing the users which were added in setup. But the catch is `load_configuration` is also called in `teardownclass`, by the time, it reaches there, the users are already removed in `teardown` and because of which there is **Unauthorized request** in `teardownclass`. 

As discussed offline, I feel `save_configuration` and `load_configuration` are API's as per their names i.e. related to configuration changes. Adding users is kind of a pre-requisite which we are doing based on our requirement.

Reply:
" Adding users is kind of a pre-requisite."
Well, I don't agree on this as this is not pre-requisite, those mgrs and oprs are needed by save/load configuration only... so we should set and reset in those methods only rather than asking the user to do before using those methods. I understand there will be overhead to set and reset mgrs and oprs but we will get save/load config method fully independent of anything else, and a user can use without any headache.

Reply:
I may have to add managers and operators in teardownclass before load_configuration and remove them again after the configuration is loaded. This seems to me as a bit of overhead. 

Please correct me if I am wrong. With my current changes, the test execution user is added as a manager role in the `setupclass` of `PBSTestSuite` if used with `--use-current-setup`. If the user is using `save/load configuration` in the test case, then the user shall still be permitted to use the API's.

Reply:
"Please correct me if I am wrong. With my current changes, the test execution user is added as a manager role in the setupclass of PBSTestSuite if used with --use-current-setup. If the user is using save/load configuration in the test case, then the user shall still be permitted to use the API's."

Yes anybody (within PTL fw + tests) can use save and load conf method. and that's why I am suggesting that we should make those methods independent of anything.

I understand there will be overhead, but everything comes with a price :smile: 

Or try to modify save and load conf methods such a way that it doesn't need mgrs and opers set? like use sudo or root user when doing privilege operation inside save/load conf methods

Reply:
okay how about this - Let's make load_configuration and save_configuration work with sudo privileges. The managers and operators will still be needed for the test execution to work properly so that can be added independent of these API's maybe after the configuration is saved. Sounds good?